frontend/model/contracts/voting/rules.js

```diff
@@ -73,3 +89,49 @@ const rules = {
 export default rules
 
 export const ruleType = unionOf(...Object.keys(rules).map(k => literalOf(k)))
+
+/**
+ *
+ * @example ('disagreement', 2, 1) => 2
+ * @example ('disagreement', 5, 1) => 3
+ * @example ('disagreement', 7, 10) => 7
+ * @example ('disagreement', 20, 10) => 10
+ *
+ * @example ('percentage', 0.5, 3) => 0.5
+ * @example ('percentage', 0.1, 3) => 0.33
+ * @example ('percentage', 0.1, 10) => 0.2
+ * @example ('percentage', 0.3, 10) => 0.3
+ */
+export const getThresholdAdjusted = (rule, threshold, groupSize) => {
+  const groupSizeVoting = Math.max(3, groupSize) // 3 = minimum groupSize to vote
+
+  return {
+    [RULE_DISAGREEMENT]: () => {
+      // Limit number of maximum "no" votes to group size
+      return Math.min(groupSizeVoting - 1, threshold)
+    },
+    [RULE_PERCENTAGE]: () => {
+      // Minimum threshold correspondent to 2 "yes" votes
+      const minThreshold = 2 / groupSizeVoting
+      return Math.max(minThreshold, threshold)
+    }
+  }[rule]()
+}
+
+/**
+ *
+ * @example (10, 0.5) => 5
+ * @example (3, 0.8) => 3
+ * @example (1, 0.6) => 2
+ */
+export const getCountOutOfMembers = (groupSize, decimal) => {
+  const minGroupSize = 3 // when group can vote
+  return Math.ceil(Math.max(minGroupSize, groupSize) * decimal)
+}
+
+export const getPercentFromDecimal = (decimal) => {
+  // convert decimal to percentage avoiding weird decimals results.
+  // e.g. 0.58 -> 58 instead of 57.99999
+  // based on https://stackoverflow.com/a/11832950/4737729
+  return Math.round(decimal * 100 * 100 / 100)
```

Comment:
This seems quite different from what's done in https://stackoverflow.com/a/11832950/4737729 ?

And `100 / 100` = `1`...

Reply:
ah, the fn started as a different thing but then I simplified it and forgot to remove that redundancy. fixed.